### DISCUSSION TITLE
fivews pattern

### DISCUSSION THREAD

**Discussion by balteravishay** [FirelyTeam/firely-net-sdk#1901] [Q&A]
Is there a way to get [fivews](http://hl7.org/fhir/fivews.html) mappings for types using the SDK? 
or at least to which property the "who" maps to in FHIR resource type.

**Comment by ewoutkramer** [comment]
Not currently.

But what would we need to do to introduce that mapping into our POCO's? If I understand this correctly there are two possibilities:

* Adding metadata in the form of reflection attributes on properties of the classes to indicate the mappings. (then you could lookup which property is the "who")
* Introducing a FiveWs `interface` and mapping the existing elements to it. (then you could just use the interface to get the "who", but you don't know which element is the "who").

Example of the first:

```csharp
[FhirType("Account","http://hl7.org/fhir/StructureDefinition/Account", IsResource=true)]
public partial class Account : Hl7.Fhir.Model.DomainResource
{
      // snip
      [FhirElement("subject", InSummary=true, Order=130)]
      [References("Patient","Device","Practitioner","Location","HealthcareService","Organization")]
      // this would be a new attribute we need to generate
      [FiveWs(Category.Subject)]
     public Hl7.Fhir.Model.ResourceReference Subject
     {
       get { return _Subject; }
       set { _Subject = value; OnPropertyChanged("Subject"); }
     }
}
```

Example of the second:

```csharp
[FhirType("Account","http://hl7.org/fhir/StructureDefinition/Account", IsResource=true)]
// Would now implement FiveW's
public partial class Account : Hl7.Fhir.Model.DomainResource : IFiveWs
{
      // snip
      [FhirElement("subject", InSummary=true, Order=130)]
      [References("Patient","Device","Practitioner","Location","HealthcareService","Organization")]
      // this would be a new attribute we need to generate
      [FiveWs(Category.Subject)]
     public Hl7.Fhir.Model.ResourceReference Subject
     {
       get { return _Subject; }
       set { _Subject = value; OnPropertyChanged("Subject"); }
     }

    // Map the current Subject to the FiveWs subject
    Hl7.Fhir.Model.ResourceReference IFiveWs.Subject => Subject;
}
```

Ideas?  What would you want to do with it?

**Reply by balteravishay** [reply]
thanks for the response, the first option seems to be consistent with how the ClassMapping/PropertyMapping work today and was where i was expecting to find such a metadata.
This mapping is very useful for generics over FHIR resources. 
another useful metadata in that sense would be the FHIR level of resource (level 4: record keeping, level 3: entities, level 2: support implementation/cross cutting, level 1: base)

**Reply by ewoutkramer** [reply]
I assume this PR has to do with this discussion ;-)
